src/map/mob.cpp

```diff
@@ -1702,8 +1702,10 @@ static bool mob_ai_sub_hard(struct mob_data *md, t_tick tick)
 
 	md->last_thinktime = tick;
 
-	if (md->ud.skilltimer != INVALID_TIMER)
+	if (md->ud.skilltimer != INVALID_TIMER) {
+		md->attacked_id = md->norm_attacked_id = 0;
```

Comment:
I'm not sure this is where we want to put this.
This means the mob drops the target while casting? Maybe we should do it at the end of `skill_castend_id`, if the cast fails and it's a mob, set the attacked_id and norm_attacked_id to 0?

Or, can we put this in mob_unlocktarget?